### DISCUSSION TITLE
Why can't we set vector type parameters from strings via Parameter, only ParameterApplicator?

### DISCUSSION THREAD

**Discussion by timsnyder** [sparcians/map#491] [Ideas]
in `template<class ValueType> template<class T, class C1> inline std::enable_if<sparta::is_vector<C1>::value>::type sparta::Parameter<ValueType>::setValueFromString_(const std::__cxx11::string &str, bool poke = false)`
at 

https://github.com/sparcians/map/blob/ee08219c9f3bf395c648720e0fdada4eb4b0c06f/sparta/sparta/simulation/Parameter.hpp#L2317-L2319

Why throw?

Couldn't we just move the code from:
https://github.com/sparcians/map/blob/ee08219c9f3bf395c648720e0fdada4eb4b0c06f/sparta/sparta/app/ConfigApplicators.hpp#L331-L355

into `Parameter` instead of throwing an exception and this would enable programmatically taking a string with a nested vector representation and setting a parameter with it just like we do via cmdline `-p`?

In other words, should the logic for YAML parsing the string and applying it to N-dimensional Vector move into `Parameter` and remove the `if if(false == p->isVector())`  from `ParameterApplicator`.

**Comment by timsnyder** [comment]
Well, the first reason I'm encountering is that simply moving the use of `sparta::ConfigParser::YAML::EventHandler` into `Parameter.hpp` creates a circular include because ConfigParserYAML.hpp includes Parameter.hpp because it was written to populate a `ParameterTree`
during the parsing.  In the case of `ParameterApplicator`, the `ptree` passed into the handler is thrown away (because it doesn't really make sense to assign a full yaml tree to `-p <some node path>`.

It seems like `sparta::ConfigParser::YAML::EventHandler` would need to be reorganized a bit so that a specialization that only parses legal parameter values (but not keys or an entire tree) can be used from `Parameter` without causing the circular includes.
